sunsynk/sensor.py

```diff
@@ -241,17 +241,72 @@ def update_value(self) -> None:
             _LOGGER.error("Could not decode temperature: %s", err)
 
 
+@attr.define(slots=True)
 class TimeRWSensor(RWSensor):
     """Extract the time."""
 
+    min: TimeRWSensor = attr.field(default=None)
+    max: TimeRWSensor = attr.field(default=None)
+
+    def available_values(self, step_minutes: int) -> List[str]:
+        """Get the available values for this sensor."""
+        full_day = 24 * 60
+
+        def reg_to_minutes(reg_value: int) -> int:
+            hours, mins = self.__reg_to_timespan(reg_value)
+            return hours * 60 + mins
+
+        min_val = reg_to_minutes(self.min.reg_value[0]) if self.min else 0
+        max_val = reg_to_minutes(self.max.reg_value[0]) if self.max else full_day
+        val = reg_to_minutes(self.reg_value[0])
+
+        time_range = self.__time_range(min_val, max_val, val, step_minutes, full_day)
+
+        return list(
+            map(lambda minutes: self.__format(*divmod(minutes, 60)), time_range)
```

Comment:
What about a small helper class for time to hide all these conversion complexities? Its also easier to unit test that way

You can store the value as (total)minutes (even mins,hrs if you wish) and have a property for regval and one for str?

```python
class ssTime:
  minutes: int
   
  @property
  def regval(self):
      return ..self.minutes...
       
  @regval.setter
  def regval(self, rval):
      self.minutes = ...rval...
```

Reply:
The helper class might not be the correct approach (since the .value / .reg_value is used by inherited methods), but you can potentially still use property get/setter to do the conversions

Reply:
:+1: Lets stay with the helper, since you've implemented it

Reply:
PAscal case would just be `SsTime` - starts with caps